tasks/utils.py

```diff
@@ -576,3 +576,22 @@ def timed(name="", quiet=False):
         res.duration = time.time() - start
         if not quiet:
             print(f"{name} completed in {res.duration:.2f}s")
+
+
+def clean_nested_paths(paths: list[str]) -> list[str]:
+    """
+    Clean a list of paths by removing paths that are included in other paths.
+
+    Example:
+    >>> clean_nested_paths(["./pkg/utils/toto", "./pkg/utils/", "./pkg", "./toto/pkg", "./pkg/utils/tata"])
+    ["./pkg", "./toto/pkg"]
+    """
+    # sort the paths by length, so that the longest paths are at the beginning
+    paths.sort()
+    cleaned_paths = []
+    for path in paths:
+        # if the path is already included in another path, skip it
+        if len(cleaned_paths) == 0 or not path.startswith(cleaned_paths[-1]):
```

Comment:
not sure we can face the case here, but if you have `./comp` and `./component` in your path list, the latter one will be cleaned-up.

Reply:
💥 You are right this would break my function. I will add a test for this case and update the function accordingly. Nice catch! Thanks!